src/Transports.AspNetCore/WebSockets/SubscriptionList.cs

```diff
@@ -0,0 +1,228 @@
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// A list of subscriptions represented by <see cref="IDisposable"/> objects.
+/// All members are thread-safe.
+/// <br/><br/>
+/// Upon signalling the cancellation token, all future calls to any method except
+/// <see cref="Dispose"/> will throw an <see cref="OperationCanceledException"/>.
+/// <br/><br/>
+/// It is recommended to call <see cref="Dispose"/> immediately after signalling
+/// the cancellation token.
+/// </summary>
+public sealed class SubscriptionList : IDisposable
+{
+    private readonly CancellationToken _cancellationToken;
+    private readonly Dictionary<string, IDisposable> _subscriptions = new();
+    private readonly object _lock = new();
+
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    public SubscriptionList(CancellationToken cancellationToken)
+    {
+        _cancellationToken = cancellationToken;
+    }
+
+    /// <summary>
+    /// Disposes of all active subscriptions.
+    /// </summary>
+    public void Dispose()
```

Comment:
If one calls `Dispose` (without signalling CancellationToken outside) then this class is still functioning.

Reply:
Yup. This class is a bit weird having both a cancellation token and dispose to shut it down, and certainly abnormal. I was thinking maybe I should implement a linked cancellation token so that dispose can signal it. Another option is to register a call to dispose from the cancellation token. I’ll think about it some more today.

Reply:
I worked on this for a while and ended up simply adding dispose checks.

Reply:
Note that I do not want to remove the cancellation token because it is important for an `OperationCanceledException` thrown under normal circumstances.  If you want to re-review this decision, we should do so during the review of #810

Reply:
As noted above, I have determined that the cancellation token is not necessary, and as such was removed from the class.